Merge SAP portal branch (cap-portal)

polybuildr opened the pull request.

polybuildr committed:
Prototype of student ambassador program registration page

polybuildr committed:
Merge pull request #50 from polybuildr/cap-portal

Prototype of student ambassador program registration page

polybuildr committed:
Merge pull request #51 from erilyth/cap-portal

Add styling to register page

polybuildr committed:
Fix scss-lint issues

polybuildr committed:
Merge pull request #52 from polybuildr/cap-portal

Fix scss-lint issues

polybuildr committed:
Add more styling to SAP registration form (and some bug fixes)

polybuildr committed:
Update SAP view

polybuildr committed:
Merge pull request #53 from polybuildr/cap-portal

Add more styling to SAP registration form (and some bug fixes)

polybuildr committed:
Add CSRF token to SAP registration form

Also create and use a csrf token library (csrf_lib).

Commit:
Add styling to register page

polybuildr committed:
Improve http error pages

Check if dedicated error page exists and load that. If it doesn't, load
a common error page and output the error code and message inside that.

Styling of error page is currently arbitrary - can be improved when rest
of the site is styled.

hharchani committed:
Merge branch 'cap-portal' of https://github.com/polybuildr/felicity16

hharchani committed:
Merge pull request #54 from polybuildr/cap-portal

Add CSRF token to SAP registration form

hharchani committed:
Send email after successful sap registration

Resolves T93

polybuildr committed:
Merge pull request #60 from hharchani/master

Send email after successful sap registration

polybuildr committed:
Merge branch '40x' into cap-portal

Conflicts:
	src/app/config-sample.php

polybuildr committed:
Remove HTTP error 400 view

Since the HTTP library handles this using a generic view now, the 400
view is no longer necessary.

hharchani committed:
Merge pull request #63 from polybuildr/cap-portal

Update handling of HTTP error pages

hharchani committed:
Use the same csrf token if available in session

polybuildr committed:
Merge pull request #64 from hharchani/change-csrf-lib

Use the same csrf token if available in session

polybuildr committed:
Split up sap::index method to improve code quality

polybuildr committed:
Use === instead of == to compare bools in sap controller

polybuildr committed:
Replace https://felicity.iiit.ac.in/ with base_url() in views

polybuildr committed:
SAP: Update <link>'s href to correct URL

The previous URL relied on a hacky production setup, this updates it to
use the actual URL.

polybuildr commented:
@erilyth could you please review the changes made from b069ddb - 4a67344 (inclusive) to just confirm that I didn't make any stupid mistakes?

erilyth commented:
Everything in b069ddb - 4a67344 seems fine

nisargjhaveri commented on sql/create_table_sap_ambassadors.sql:
```diff
@@ -0,0 +1,65 @@
+  `about_you` text NOT NULL,
+  `organised_event` text,
+  `registration_time` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
We should make effort to always use utf8 charset.

nisargjhaveri commented:
Scrutinizer still have some patches for code styling.

polybuildr commented:
I looked at the patches for all files introduced in this PR.

Apart from the `if (! isset` (the space), all the other patches don't seem valid. And the `isset` one was intentional, so I didn't make that change either.

nisargjhaveri commented:
https://scrutinizer-ci.com/g/felicity-iiith/felicity16-website/inspections/744fd752-7c02-49e0-9f9f-ad5e1a91e211/patches?status=new 

Apart from the first one everything seems valid.

polybuildr committed:
Fix minor indentation and spacing code style issues

polybuildr committed:
Update sql/: add a full schema and also a migrations/ directory

We can keep the complete up-to-date schema in sql/*.sql. We can keep
delta migrations in sql/migrations/*.sql. More thought needs to be given
to this, but it seems alright for now.

nisargjhaveri commented:
Add the final schema somewhere?

polybuildr commented:
Yes, adding.

Vivek Ghaisas

On Thu, Dec 17, 2015 at 10:05 AM, Nisarg Jhaveri notifications@github.com
wrote:

> Add the final schema somewhere?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/felicity-iiith/felicity16-website/pull/65#issuecomment-165363837
> .

nisargjhaveri commented:
Looks fine. Merging.

polybuildr commented:
Finally! :D
